lib/rules/no-duplicate-landmark-labels.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Duplicate label on <${element}> detected. If two of the same landmark elements or landmark roles are found in the same template, ensure that they have a unique label (provided by the aria-label or aria-labelledby attribute).`;
+}
+
+const ROLE_LANDMARK_MAP = {
+  banner: 'header',
+  main: 'main',
+  complementary: 'aside',
+  form: 'form',
+  search: 'form',
+  navigation: 'nav',
+  contentinfo: 'footer',
+};
+
+const LANDMARK_ELEMENTS = ['header', 'main', 'aside', 'form', 'main', 'nav', 'footer'];
+
+const LANDMARK_ROLES = Object.keys(ROLE_LANDMARK_MAP);
+
+module.exports = class NoDuplicateLandmarkLabels extends Rule {
+  constructor(options) {
+    super(options);
+    this._landmarkHash = [];
+  }
+
+  visitor() {
+    return {
+      ElementNode(node) {
+        const hasRoleAttribute = AstNodeInfo.hasAttribute(node, 'role');
+
+        // Return if it is not a landmark element or doesn't have role
+        if (!(hasRoleAttribute || LANDMARK_ELEMENTS.includes(node.tag))) {
+          return;
+        }
+
+        const roleValue = AstNodeInfo.elementAttributeValue(node, 'role');
```

Comment:
The code below doesn't seem to account for `roleValue` to be anything other than a string, but in reality it can be either a string or AST node.

Reply:
You mean something like this:
```js
<div role={{landmarkRole}}></div>
```

Reply:
Yes, exactly.

Reply:
FWIW, I've tracked removing this "automatic unwrapping" from `ASTNodeInfo.elementAttributeValue` in https://github.com/ember-template-lint/ember-template-lint/issues/1358.

Reply:
TODO: update this code to reflect that it could be something other than a string